plugins/modules/ece_cluster_alias.py

```diff
@@ -0,0 +1,108 @@
+        deployment_name=dict(type='str'),
+        deployment_id=dict(type='str', default=None),
+        no_cluster_object=dict(type='bool', default=True),
+        alias_name=dict(type='str', required=True)
+    )
+    argument_dependencies = []
+        #('state', 'present', ('enabled', 'alert_type', 'conditions', 'actions')),
+        #('alert-type', 'metrics_threshold', ('conditions'))
+    
+    module = AnsibleModule(argument_spec=module_args, required_if=argument_dependencies, supports_check_mode=True)
+    
+    deployment_name = module.params.get('deployment_name')
+    deployment_id = module.params.get('deployment_id')
+    alias_name = module.params.get('alias_name')
+    results = { 'changed': True }
+        
+    ElasticDeployments = ECE(module)
+    
+    if deployment_id:
+      deployment_object = [ElasticDeployments.get_deployment_byid(deployment_id)]
+    elif deployment_name:
+      deployment_object = [ElasticDeployments.get_deployment_info(deployment_name)]
+      
+    if len(deployment_object) == 1:
+      deployment_object = deployment_object[0]
+      update_body = {
+        'alias': alias_name,
+        'prune_orphans': False,
+        'resources': {
+          'elasticsearch': [
+            {
+              'region':  deployment_object['resources']['elasticsearch'][0]['region'],
+              'ref_id':  deployment_object['resources']['elasticsearch'][0]['ref_id'],
+              'plan': deployment_object['resources']['elasticsearch'][0]['info']['plan_info']['current']['plan']
+            }
+          ],
+          'kibana': [
+            {
+              'region':  deployment_object['resources']['kibana'][0]['region'],
+              'ref_id':  deployment_object['resources']['kibana'][0]['ref_id'],
+              'elasticsearch_cluster_ref_id':  deployment_object['resources']['elasticsearch'][0]['ref_id'],
+              'plan': deployment_object['resources']['kibana'][0]['info']['plan_info']['current']['plan']
+            }
+          ]
+        }
+      }
+      
+      ElasticDeployments.update_deployment_byid(deployment_object['id'], update_body)
+      
+      deployment_healthy = ElasticDeployments.wait_for_cluster_state(deployment_object['id'], "elasticsearch" )
+      deployment_healthy = ElasticDeployments.wait_for_cluster_state(deployment_object['id'], "kibana" )
+      deployment_healthy = ElasticDeployments.wait_for_cluster_state(deployment_object['id'], "kibana","main-apm")
+      
+      if deployment_healthy == False:
+        results['cluster_alias_status'] = "Cluster information may be incomplete because the cluster is not healthy"
+      else:
+        time.sleep(30)
+        
+      results['changed'] = True
+    else:
+      results['changed'] = False
+      results['cluster_alias_status'] = "0 or more than 1 deployment was matched with the name " + deployment_name + " or id " + deployment_id
+    module.exit_json(**results)
+
+if __name__ == "__main__":
+    main()
+    
+     

```

Comment:
So the basic design theory is that we want to enable playbooks to be declarative expressions of the intended config rather than collections of tasks that are just being executed with a different interface.  In an ideal world we want a playbook to read like a config file with some variables templated in, that's not always possible but provides a good way of looking at design decisions.

In this case - the alias is a property of the cluster, and if we were just defining the setup of a cluster in a static config file it would just be line in that file, and so it probably belongs as part of the cluster module.  It generally shouldn't be an issue for us to re-run the module against an existing cluster as long as we've implemented idempotency correctly so that part shouldn't be a major issue here (although may need to be careful if there are defaults set for sizing in the module)